clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -494,7 +492,8 @@ ProgramInfo::insertIntoExternalFunctionMap(ExternalFunctionMapType &Map,
         (newC->hasBody() ||
          (oldC->numParams() == 0 && newC->numParams() != 0))) {
       newC->brainTransplant(oldC, *this);
-      Map[FuncName] = ToIns;
+      Map.erase(FuncName);
+      Map.insert(std::make_pair(FuncName, ToIns));
```

Comment:
Same question as above -- why is this code change material/useful ?

Reply:
The specific reason for having `erase` here and not above is that the defualy behavior of `insert` is noop if the key is already in the map, while `operator[]` will replace any existing entries. In this case, this is the only case where the key is already in the map. As above, I think I can revert this after removing all the constraint variable sets.